lib/mapping/MapFormatH3M.cpp

```diff
@@ -1842,77 +1854,89 @@ void CMapLoaderH3M::readSeerHutQuest(CGSeerHut * hut, const int3 & position)
 
 	if(hut->quest->missionType)
 	{
-		auto rewardType = static_cast<CGSeerHut::ERewardType>(reader->readUInt8());
-		hut->rewardType = rewardType;
+		auto rewardType = reader->readUInt8();
+		Rewardable::VisitInfo vinfo;
+		auto & reward = vinfo.reward;
 		switch(rewardType)
 		{
-			case CGSeerHut::EXPERIENCE:
+			case 0: //NOTHING
 			{
-				hut->rVal = reader->readUInt32();
+				// no-op
 				break;
 			}
-			case CGSeerHut::MANA_POINTS:
+			case 1: //EXPERIENCE
 			{
-				hut->rVal = reader->readUInt32();
+				reward.heroExperience = reader->readUInt32();
 				break;
 			}
-			case CGSeerHut::MORALE_BONUS:
+			case 2: //MANA POINTS:
 			{
-				hut->rVal = reader->readUInt8();
+				reward.manaDiff = reader->readUInt32();
 				break;
 			}
-			case CGSeerHut::LUCK_BONUS:
+			case 3: //MORALE_BONUS
 			{
-				hut->rVal = reader->readUInt8();
+				reward.bonuses.emplace_back(BonusDuration::ONE_BATTLE, BonusType::MORALE, BonusSource::OBJECT, reader->readUInt8(), hut->id);
 				break;
 			}
-			case CGSeerHut::RESOURCES:
+			case 4: //LUCK_BONUS
 			{
-				hut->rID = reader->readUInt8();
-				hut->rVal = reader->readUInt32();
-
-				assert(hut->rID < features.resourcesCount);
-				assert((hut->rVal & 0x00ffffff) == hut->rVal);
+				reward.bonuses.emplace_back(BonusDuration::ONE_BATTLE, BonusType::LUCK, BonusSource::OBJECT, reader->readUInt8(), hut->id);

```

Comment:
Likely - `hut->id` is also uninitialized

Reply:
not sure how to handle it properly

Reply:
ObjectInstanceID is present in method that calls this one (or someplace higher in the call stack)